### PR TITLE
fix(archive): Improve archive preference handling

### PR DESCRIPTION
**Pull request by nickvergessen**
### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not possible
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/Service/ParticipantService.php [hunk 1]
```diff
@@ -299,18 +299,36 @@ public function updateNotificationCalls(Participant $participant, int $level): v
...
+			if ($attendee->getNotificationCalls() === Participant::NOTIFY_CALLS_ON) {
+				$attendee->setNotificationCalls(Participant::NOTIFY_CALLS_OFF);
+			}
+
+			// Reduce notifications when it was "Always" via the default
+			// Otherwise we respect the attendees selection
+			if ($attendee->getNotificationLevel() === Participant::NOTIFY_DEFAULT
+				&& $this->getDefaultGroupNotification() !== Participant::NOTIFY_NEVER) {
+				$attendee->setNotificationCalls(Participant::NOTIFY_MENTION);
+			}
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply lib/Service/ParticipantService.php:326]: I assume DEFAULT is stored in DB per each attendee, unless was redefined by that attendee. Then it makes sense to redefine default behaviour only

[the remark this answers is not in the document]
Comment (DorraJaouad) [reply lib/Service/ParticipantService.php:326]: > I assume DEFAULT is stored in DB per each attendee, unless was redefined by that attendee. Then it makes sense to redefine default behaviour only

Even if it was redefined by attendee, it doesn't mean it should be applied in archive (the point of archive it to put the notifications frequency to the lowest (not in sidebar list, no call, no message notif). 

> If they are annoyed by notifications, maybe they shouldn't have set it to ALWAYS?)

Conversation was previously important (a deal) and now it's not (deal closed for a while now). A user can see archiving as a good shortcut to reduce the notifications at once.

**Comment by nickvergessen** [comment]
Replaced by https://github.com/nextcloud/spreed/pull/13814

**Pull request closed without merging**